cvmfs/sync_mediator.cc

```diff
@@ -94,6 +95,33 @@ void SyncMediator::Add(SharedPtr<SyncItem> entry) {
     return;
   }
 
+  // .cvmfsbundles file type
+  if (entry->IsBundleSpec() && entry->IsRegularFile() &&
```

Comment:
We should be more strict here: with this code, if we have a "weird" .cvmfsbundle file we'll ignore the special name.  However, we should check for `IsBundleSpec()` and if it is weird (no regular file, hardlink, etc.) then we abort.